File: 0056-merge-intervals/0056-merge-intervals.ts
function merge(intervals: number[][]): number[][] {
    if (intervals.length == 1) return intervals;
    intervals.sort((a, b) => (a[0] - b[0]));
    const res = [];
    console.log(intervals);
    let current = intervals[0];
    for (let i = 1; i < intervals.length; i++) {
        const [start, end] = intervals[i];
        if (current[1] < start) {
            res.push(current);
            current = intervals[i];
        } else {
            current[1] = Math.max(end, current[1]);
        }
    }
    res.push(current);
    return res;
};

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        int[] currentInterval = intervals[0];

        List<int[]> res = new ArrayList<>();

        for (int i = 1; i < intervals.length; i++) {
            int[] interval = intervals[i];

            if (interval[0] <= currentInterval[1]) {
                currentInterval[1] = Math.max(currentInterval[1], interval[1]);
            } else {
                res.add(currentInterval);
                currentInterval = interval;
            }
        }
        res.add(currentInterval);

        return res.toArray(new int[res.size()][]);
    }
}